ci/etcd.rb

```diff
@@ -0,0 +1,51 @@
+require './ci/common'
+
+namespace :ci do
+  namespace :etcd do |flavor|
+    task :before_install => ['ci:common:before_install']
+
+    task :install => ['ci:common:install'] do
+      sh %(curl -s -L -o $VOLATILE_DIR/etcd.tar.gz\
+            https://github.com/coreos/etcd/releases/download/v2.0.3/etcd-v2.0.3-linux-amd64.tar.gz)
+      sh %(mkdir $VOLATILE_DIR/etcd)
+      sh %(tar xzvf $VOLATILE_DIR/etcd.tar.gz\
+                    -C $VOLATILE_DIR/etcd\
+                    --strip-components=1 >/dev/null)
+    end
+
+    task :before_script => ['ci:common:before_script'] do
+      sh %(cd $VOLATILE_DIR/etcd && ./etcd >/dev/null &)
+      sleep_for 10
+    end
+
+    task :script => ['ci:common:script'] do
+      this_provides = [
+        'etcd'
+      ]
+      Rake::Task['ci:common:run_tests'].invoke(this_provides)
+    end
+
+    task :cleanup => ['ci:common:cleanup'] do
+      sh %(rm -rf $VOLATILE_DIR/etcd* || echo 'etcd not running')
```

Comment:
You don't need to clean up `VOLATILE_DIR` it's already taken care of by the common cleanup task.
However, you might want to kill the etcd daemon you launched in the `before_script` stage, either by using a pidfile if you can have one or by using an appropriate kill command.